### PR TITLE
Intercept query param for <Mentoring.Inbox />

### PR DESCRIPTION
**Pull request by kntsoriano**
https://github.com/exercism/website/issues/753

@iHiD Can you check if this works well for you? Should we start debouncing the text fields to prevent a push state every time the user types on the text field?

If this works well, I will add this feature to our other components.

### CODE REVIEW

**Comment by iHiD** [comment]
Nice! I think this is really good!!

@ErikSchierboom Could you check it out and see what you think too. Click around in the mentoring inbox, then go to a different page, then use the back button and it should take you through the relevant tabs/pages etc.

**Comment by iHiD** [comment]
@SleeplessByte Any comments before we roll this out next week?

**Comment by SleeplessByte** [comment]
Instead of pushing each change, I would only push after a _debounce_, and replaceState otherwise.

File: app/javascript/hooks/use-history.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      if (typeof v === 'string') {
+        return v.length > 0
+      } else {
+        return v !== null
+      }
+    })
+    .reduce((acc, [k, v]) => ({ ...acc, [k]: v }), {})
+}
+
+export const pushState = (params: any) => {
```
Comment (SleeplessByte) [inline app/javascript/hooks/use-history.ts]: Are these purposefully typed as `any`?

Comment (kntsoriano) [reply app/javascript/hooks/use-history.ts]: I've added generic parameters now.

**Comment by kntsoriano** [comment]
@SleeplessByte @iHiD I've updated it now to debounce before pushing to the history.

**Comment by kntsoriano** [comment]
@iHiD I'm pretty happy with this already. I will merge this in once tests pass.

**Comment by iHiD** [comment]
OK great. Merged! :)

**Pull request merged**